src/gleam/int.gleam

```diff
@@ -27,19 +27,22 @@ pub fn absolute_value(x: Int) -> Int {
 ///
 /// ```gleam
 /// > power(2, -1.0)
-/// 0.5
+/// Ok(0.5)
 ///
 /// > power(2, 2.0)
-/// 4
+/// Ok(4.0)
 ///
 /// > power(8, 1.5)
-/// 22.627416997969522
+/// Ok(22.627416997969522)
 ///
 /// > 4 |> power(of: 2.0)
-/// 16.0
+/// Ok(16.0)
+///
+/// > power(-1, 0.5)
+/// Error(Nil)
 /// ```
 ///
-pub fn power(base: Int, of exponent: Float) -> Float {
+pub fn power(base: Int, of exponent: Float) -> Result(Float, Nil) {
```

Comment:
Perhaps exponent should also be an int. That would make this function able to return int rather than result, right?

Reply:
Depending on what should be returned when you try `power(0, -1) == 1 / 0 ` it is possible to return just an int.

Reply:
Sorry, I meant just a float, but you have answered my question. Let's keep it as is, your design is great

Reply:
I like this, that means the int power function gets and returns ints, right?

The docblock could take a notice that float.power exists if you need negative or fractional exponents I suppose?

Reply:
Hmm yeah it could work like that, but now that I think about it again you'll probably need to return an error to the user if they try to give a negative exponent : /

Reply:
That is something like this:

```gleam
pub fn power(base: Int, of exponent: Int) -> Result(Int, Nil) {
  case exponent >= 0 {
      True -> {
        assert Ok(result) = float.power(to_float(base), exponent)
        result
        |> float.round()
        |> Ok
      }
      False -> Error(Nil)
    }
}
```

Reply:
Another option is to make the behaviour so that if the exponent is negative then instead use 0 as the exponent. There is precedence for this in functions like list.take where if the number is negative we use 0 there too rather than erroring. It would need to be clearly documented.

Reply:
IMHO this case is different than both `1/0` and `list.take()`.

- Devs (should) always know that `1/0` does something unexpected so they should inform themselves what Gleam does.
- Lists cannot have negative indexes (unless you define that to be looping from the back)
- A negative integer as an exponent is a totally reasonable option.

For my opinion weights little, my 2 cents: Do not create unexpected behavior (this would be) and instead simply use Results. It is easy to use unwrap, try or assert IMHO. The pain is little, the gain is no magick hidden in implementation requiring to read doc blocks or the implementation itself.

Reply:
In that case, maybe leaving the `int.power` implementation as it is right now is probably the most intuitive.